prisma/schema.prisma

```diff
@@ -110,3 +110,38 @@ model Comment {
   replies   Comment[] @relation("Replies")
   replyToId Int?
 }
+
+model AuctionProfile {
+  id       Int              @id @default(autoincrement())
+  name     String           @unique
+  email    String           @unique
+  avatar   String?
+  password String
+  salt     String
+  credits  Int
```

Comment:
Good catch. I did find [this issue](https://github.com/prisma/prisma/issues/12969) that could be related. Seems like Prisma will return an error which we should handle in some way.

Reply:
Setting my credits to max of int4 and winning a listing will result in a server crash as it throws an error. We'll need to handle this in some way for sure. Not sure what you think is best?

Reply:
Can we catch that error?

I would set the profile credits to 0.

Reply:
I'll have a closer look again later. This would probably have to be done inside the listing.utils in some way, just not sure what the best way to tackle this is.

Reply:
Could solve this by wrapping in try/catch and resetting credits to 0. We probably want to handle this in a similar way for the `Promise.all(loser.map())` too. 

What do you think of something like this?
```js
// Transfer credits from highest bid to seller
try {
  await prisma.auctionProfile.update({
    where: { name: listing.sellerName },
    data: { credits: { increment: winner.amount } }
  })
} catch (err) {
  if (err instanceof Prisma.PrismaClientUnknownRequestError) {
    await prisma.auctionProfile.update({
      where: { name: listing.sellerName },
      data: { credits: 0 }
    })
  }
}
```

Reply:
We could set an arbitrary maximum and just cap credits there. Like 1,000,000 + 1 = 1,000,000

But the above looks like it would work fine

Reply:
Unfortunately there's no way for Prisma to solve this, at least natively without . Would be great for them to have an attribute for it. I found this old [ref](https://github.com/prisma/prisma1/issues/728).
```sql
credits  Int  @max(100000)
credits  Int  @constraint(min: 0, max: 1000000)
```
There are other packages for it, for example [graphql-constraint-directive](https://github.com/confuser/graphql-constraint-directive).

Personally I like capping it to a max and just not let it increase. Open to a suggestion of how to implement it correctly though.

Reply:
Looks good! Worth also using `awardCreditsOrCap` inside the `Promise.all()`? Could those realistically ever overflow seeing as you'll only get back what you already spent?

Reply:
Yeah I don't think it will hurt. I'll change that now

Reply:
Thank you!